packages/components/src/Actions/ActionButton/ActionButton.component.js

```diff
@@ -206,7 +206,8 @@ function ActionButton(props) {
 		);
 	}
 
-	if (hideLabel || tooltip || tooltipLabel) {
+	const hasLabel = tooltipLabel || label;
+	if (hasLabel && ((hideLabel && tooltip !== false) || tooltip)) {
```

Comment:
It's not enough?
```suggestion
	if (hasLabel && (tooltip || hideLabel)) {
```

Reply:
Nope because if you set explicitly the `tooltip` as `false` with `hideLabel` as `true` on your component then you don't want the tooltip whereas in your suggestion the tooltip will be displayed. I've added a test to cover that case 👍